backend/src/test/java/de/kathahuth/gymast/controller/GroupControllerTest.java

```diff
@@ -51,9 +65,34 @@ public void getAllGroups() {
 
         //Then
         assertThat(response.getStatusCode(), is(HttpStatus.OK));
-        assertThat(response.getBody(), arrayContainingInAnyOrder(
-                new Group("difficultygroup1"),
-                new Group("difficultygroup2")));
+        assertThat(response.getBody(), arrayContainingInAnyOrder(createDynamicGroup(), createJumpGroup()));
+    }
+
+    @Test
+    @DisplayName("Get Group By Id should return GroupId")
+    public void getGroupById(){
+        //Given
+        String groupId = "Dynamic Strength";
+        groupMongoDb.save(createDynamicGroup());
+
+        //When
+        ResponseEntity<Group> response = testRestTemplate.getForEntity(getUrl() + "/" + groupId, Group.class);
```

Comment:
you could also change your `getUrl()` method to end with `/` so you do not need to include it here